sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -0,0 +1,681 @@
+
+
+COVARIANCE_TYPE = ['full', 'tied', 'diag', 'spherical']
+
+
+def generate_data(n_samples, n_features, weights, means, covariances,
+                  covariance_type):
+    rng = np.random.RandomState(0)
+
+    X = []
+    if covariance_type == 'spherical':
+        for _, (w, m, c) in enumerate(zip(weights, means,
+                                          covariances['spherical'])):
+            X.append(rng.multivariate_normal(m, c * np.eye(n_features),
+                                             int(np.round(w * n_samples))))
+    if covariance_type == 'diag':
+        for _, (w, m, c) in enumerate(zip(weights, means,
+                                          covariances['diag'])):
+            X.append(rng.multivariate_normal(m, np.diag(c),
+                                             int(np.round(w * n_samples))))
+    if covariance_type == 'tied':
+        for _, (w, m) in enumerate(zip(weights, means)):
+            X.append(rng.multivariate_normal(m, covariances['tied'],
+                                             int(np.round(w * n_samples))))
+    if covariance_type == 'full':
+        for _, (w, m, c) in enumerate(zip(weights, means,
+                                          covariances['full'])):
+            X.append(rng.multivariate_normal(m, c,
+                                             int(np.round(w * n_samples))))
+
+    X = np.vstack(X)
+    return X
+
+
+class RandomData(object):
+    def __init__(self, rng, n_samples=500, n_components=2, n_features=2):
+        self.n_samples = n_samples
+        self.n_components = n_components
+        self.n_features = n_features
+
+        self.weights = rng.rand(n_components)
+        self.weights = self.weights / self.weights.sum()
+        self.means = rng.rand(n_components, n_features) * 50
+        self.covariances = {
+            'spherical': .5 + rng.rand(n_components),
+            'diag': (.5 + rng.rand(n_components, n_features)) ** 2,
+            'tied': make_spd_matrix(n_features, random_state=rng),
+            'full': np.array([make_spd_matrix(
+                n_features, random_state=rng) * .5
+                for _ in range(n_components)])}
+
+        self.X = dict(zip(COVARIANCE_TYPE, [generate_data(
+            n_samples, n_features, self.weights, self.means, self.covariances,
+            cov_type) for cov_type in COVARIANCE_TYPE]))
+        self.Y = np.hstack([k * np.ones(int(np.round(w * n_samples)))
+                            for k, w in enumerate(self.weights)])
+
+
+def test_gaussian_mixture_parameters():
+    # test bad parameters
+    rng = np.random.RandomState(0)
+
+    # n_init should be greater than 0
+    n_init = rng.randint(-10, 1)
+    gmm = GaussianMixture(n_init=n_init)
+    assert_raise_message(ValueError,
+                         "Invalid value for 'n_init': %d "
+                         "Estimation requires at least one run"
+                         % n_init,
+                         gmm._check_initial_parameters)
```

Comment:
You shouldn't call `_check_initial_parameters` here.
You never use this function, yet your test is not able to tell you that.